api/src/main/java/com/nicknathanjustin/streamercontracts/votes/VoteServiceImpl.java

```diff
@@ -132,33 +146,45 @@ private boolean userCanVoteOnContract(@NonNull final UserModel voter, @NonNull f
             log.warn("Multiple votes detected. User: {} attempted to vote on contractId: {} more than once",
                     voter.getId(),
                     contractModel.getId());
+
             return false;
         }
+
         return true;
     }
 
-    private void validateContract(@NonNull final ContractModel contractModel, final boolean validateVote) {
-        if (contractModel.isCommunityContract()) {
-            throw new IllegalArgumentException("Voting and Settling logic not yet implemented for community contracts. ContractId " + contractModel.getId());
+    private boolean isContractVotable(@NonNull final ContractModel contractModel) {
+        UUID contractId = contractModel.getId();
+        if (!contractModel.getState().equals(ContractState.ACCEPTED.name())) {
+            log.warn(String.format("Cannot vote on a contract that has not been accepted. Contract Id: %s", contractId));
```

Comment:
These are things that shouldn't happen, but might since the endpoints are exposed. If we throw an exception its going to return a 500 error back to the user. Ideally we should catch these exception and return some user friendly message back to the user.

Reply:
"Ideally we should catch these exception and return some user friendly message back to the user." Agreed, thats what the controller is good at doing. Its the layer between our clients and our services. Its normal for a controller to call a service that may throw, catch whats thrown and return a nice message to the end user.

One thing that was nice about leaving these as exceptions is we propagated the cause of the error in the exception back to our caller. So If a user couldnt vote on something because it wasnt accepted, expired, ect... we would know that in the controller by checking the exception message. Switching to logs like this we no longer propagate the message back and can only say the contract wasnt votable. we couldnt offer a reason without altering the service to return some field that says why a vote failed.

I think exceptions were the better choice here but wont block the CR for it.

Reply:
Ok I agree with that. I can change them back to exceptions then. They're not currently being caught by the controller though, so we'll need to catch them at some point in the future

Reply:
Left as a TODO to catch the exceptions.